### PR TITLE
Minor code cleanup in Delta lake connector

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Minor code cleanup in Delta lake connector

## Additional context and related issues
NA

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findinpath** [comment]
Do address also https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergAvroDataConversion.java#L163

**Comment by krvikash** [comment]
> Do address also:
> 
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergAvroDataConversion.java#L163
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergAvroDataConversion.java#L313-L315
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergAvroDataConversion.java#L324-L327
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java#L744-L745
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java#L840-L841
> * https://github.com/trinodb/trino/blob/9a74942e0e79ae2650f8a77ab87a8bb7368bdb5e/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java#L843-L844

I have a separate PR https://github.com/trinodb/trino/pull/15488/ to use pattern matching for `instanceOf`. Also, some of the locations which you have shared are not from the master branch and have already been addressed earlier. I will take care rest of the changes in https://github.com/trinodb/trino/pull/15488/

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java [hunk 1]
```diff
@@ -558,7 +556,6 @@ private DataFileInfo copyParquetPageSource(DeltaLakeWriter fileWriter)
             if (successfulWrite) {
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:556]: Do note that the code will behave slightly differently after this change.

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

> Note: A try-with-resources statement can have catch and finally blocks just like an ordinary try statement. In a try-with-resources statement, any catch or finally block is run after the resources declared have been closed.

As I understand it, `fileWriter.commit()` statement will be executed after `connectorPageSource.close()`

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java [hunk 2]
```diff
@@ -157,12 +157,9 @@ public RowType getAddEntryType(MetadataEntry metadataEntry, boolean requireWrite
...
     {
         if (columnType instanceof RowType rowType) {
-            if (columnName.isPresent()) {
-                return RowType.field(
-                        columnName.get(),
-                        RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
-            }
-            return RowType.field(RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
+            return columnName
+                    .map(name -> RowType.field(name, RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList()))))
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java:161]: `rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())` is common for both code branches. Let's extract it to a separate variable.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java (hunk 1 above)
Comment (krvikash) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:556]: Yes, Thanks for catching this. I will revert this change.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ protected HiveMinioDataLake createHiveMinioDataLake()
         FileAttribute<Set<PosixFilePermission>> posixFilePermissions = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
         Path hadoopCoreSiteXmlTempFile = Files.createTempFile("core-site", ".xml", posixFilePermissions);
         hadoopCoreSiteXmlTempFile.toFile().deleteOnExit();
-        Files.write(hadoopCoreSiteXmlTempFile, abfsSpecificCoreSiteXmlContent.getBytes(UTF_8));
+        Files.writeString(hadoopCoreSiteXmlTempFile, abfsSpecificCoreSiteXmlContent);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:91]: There're other 2 usages of `Files.write()` in Delta Lake connector.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java [hunk 4]
```diff
@@ -157,12 +157,14 @@ public RowType getAddEntryType(MetadataEntry metadataEntry, boolean requireWrite
...
         if (columnType instanceof RowType rowType) {
-            if (columnName.isPresent()) {
-                return RowType.field(
-                        columnName.get(),
-                        RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
-            }
-            return RowType.field(RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
+            RowType rowTypeFromFields = RowType.from(
+                    rowType.getFields()
+                            .stream()
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java:162]: We generally format like this (move stream to the previous line):
```java
                    rowType.getFields().stream()
                            .map(field -> buildNullCountType(field.getName(), field.getType()))
```

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java [hunk 5]
```diff
@@ -157,12 +157,14 @@ public RowType getAddEntryType(MetadataEntry metadataEntry, boolean requireWrite
...
+                            .map(field -> buildNullCountType(field.getName(), field.getType()))
+                            .collect(toImmutableList()));
+            return columnName
+                    .map(name -> RowType.field(name, rowTypeFromFields))
+                    .orElseGet(() -> RowType.field(rowTypeFromFields));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java:167]: Why not `return new RowType.Field(columnName, rowTypeFromFields)`?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java [hunk 6]
```diff
@@ -157,12 +157,14 @@ public RowType getAddEntryType(MetadataEntry metadataEntry, boolean requireWrite
...
         }
         return columnName.map(name -> RowType.field(name, BigintType.BIGINT)).orElse(RowType.field(BigintType.BIGINT));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java:169]:
```suggestion
return new RowType.Field(columnName, BigintType.BIGINT)
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java (hunk 3 above)
Comment (krvikash) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:91]: The other 2 places do not have String value but have byte[] value.

https://github.com/trinodb/trino/blob/98b274bf9a19a11fff870e5dea486042435f2a7b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java#L89-L93

https://github.com/trinodb/trino/blob/98b274bf9a19a11fff870e5dea486042435f2a7b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/metastore/glue/TestDeltaLakeGlueMetastore.java#L269-L270

Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:91]: I didn't look into the 1st case. Regarding the 2nd case, there's `com.google.common.io.Resources#toString(URL url, Charset charset)` method.

Comment (krvikash) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:91]: Thanks for the pointer. Addressed comment.

**Review by ebyhr — APPROVED** [review]
Please squash 1st and 3rd commits into one.

**Review by findinpath — APPROVED** [review]

**Pull request merged**